File: nestedscrolldemo/src/main/java/com/example/nestedscrolldemo/adapter/MyAdapter.java
package com.example.nestedscrolldemo.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.nestedscrolldemo.IconRepo;
import com.example.nestedscrolldemo.R;
import com.example.nestedscrolldemo.bean.IconBean;

import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyHolder> {
    private List<IconBean> iconList;
    private Context mContext;

    public MyAdapter(Context context) {
        mContext = context;
        iconList = IconRepo.getIconBeanList();
    }

    @NonNull
    @Override
    public MyHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new MyHolder(LayoutInflater.from(mContext).inflate(R.layout.item_icon, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull MyHolder holder, int position) {
        IconBean icon = iconList.get(position);
        holder.ivIcon.setImageResource(icon.resId);
        holder.tvName.setText(icon.name);
    }

    @Override
    public int getItemCount() {
        return iconList.size();
    }


    class MyHolder extends RecyclerView.ViewHolder{
        ImageView ivIcon;
        TextView tvName;
        public MyHolder(@NonNull View itemView) {
            super(itemView);
            ivIcon = itemView.findViewById(R.id.iv_icon);
            tvName = itemView.findViewById(R.id.tv_name);
        }
    }
}

File: coordinaotrlayoutdemo/src/main/java/com/example/coordinaotrlayoutdemo/adapter/MyAdapter.kt
package com.example.coordinaotrlayoutdemo.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.coordinaotrlayoutdemo.R
import com.example.coordinaotrlayoutdemo.repo.UserRepo

class MyAdapter(val context : Context) : RecyclerView.Adapter<MyHolder>() {
    private val datas : List<UserRepo.User> = UserRepo.getUsers()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        return MyHolder(LayoutInflater.from(context).inflate(R.layout.item_mine, parent, false))
    }

    override fun getItemCount(): Int {
        return datas.size
    }

    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        holder.tvContent.text = datas[position].content
    }
}

class MyHolder(private val view: View) : RecyclerView.ViewHolder(view) {
    val tvContent : TextView
    init {
        tvContent = itemView.findViewById(R.id.tv_content)
    }

}